src/cache_cf.cc

```diff
@@ -621,28 +619,21 @@ parseConfigFileOrThrow(const char *file_name)
      */
     configDoConfigure();
 
+    // TODO: Throw before configDoConfigure(). Doing that would reduce the set
+    // of configuration errors Squid can detect in one execution, but we should
+    // not apply a clearly broken configuration, especially when we are going to
+    // quit anyway. While some legacy parse_foo() functions apply significant
+    // changes before configDoConfigure(), we cannot easily stop them. We can
+    // easily stop configDoConfigure().
+    if (unrecognizedDirectives)
+        throw TextException(ToSBuf("Found ", unrecognizedDirectives, " unrecognized directive(s)"), Here());
```

Comment:
I am happy to implement this TODO in this PR if others prefer that, but I think it may be handled better in another PR because we may be able to add more changes that fix similar "applying bad configuration" problems. We will not be able to completely eliminate those problems in the foreseeable future, but if we can easily squash more, we should.